### DISCUSSION TITLE
Is this only for validation?

### DISCUSSION THREAD

**Discussion by xucian** [python-jsonschema/jsonschema#1046] [General]
Hello,

Quite simple question here. Can this be used for something other than validation?
Seems not, but just wanted to ask. I'm looking for a schema generator from plain JSON data.

Thanks for the great work!

**Comment by mdavis-xyz** [comment]
I'm not the maintainer, but my guess is the answer is no.

How do you generate a schema from json? You can't tell whether the fields present are optional. Or perhaps an enum of types.

**Comment by xucian** [comment]
It's just for the bulk of it, the result is manually changed after. This
would be a huge time saver, especially for large jsons

**Comment by Julian** [comment]
This library only implements validation at the minute, yes.
